poetry/console/commands/shell.py

```diff
@@ -0,0 +1,24 @@
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</info> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        if self.venv.is_venv():
+            self.line(f"Spawning shell within <info>{self.venv.venv}</info>")
+
+            shell = self.venv.get_shell()
```

Comment:
This almost works, except on some systems, this may return `None` if the shell is not found. You should handle that case.

Also, on Windows, you should execute `cmd /k`.

Reply:
Ok, I handled the Windows case, but what should it do if it's not on Windows but `get_shell` came back `None`?

Reply:
So, `SHELL` is not really set to the current running shell. It appears to be the login shell. 
This seems to work:
```python
ps = run(["ps", "-p", str(getppid()), "-oargs="], stdout=PIPE)
shell = ps.stdout.decode().strip()
```
... but I'm not really confident about it (on mac it is a bit weird). Maybe we can just assume that the user wants the default shell and fallback to `bash` if it's not set.

About Windows, it seems more complicated than that, but I can't really test it. I was taking a look at how Pipenv handles it: https://github.com/pypa/pipenv/blob/master/pipenv/patched/pew/_win_utils.py

Reply:
Re `$SHELL` being unset: I'd give an error and ask people to set `$SHELL`. Better to have the user be explicit in what they want then be wrong in trying to guess. I think it is ok to assume people are using the login shell if available.

Also, you are correct it is a bit more complicated on Windows. I suppose powershell/cmder support would be important :) I think it is okay if you handle Unixs and I can come back to handle Windows.